openmdao/core/implicitcomponent.py

```diff
@@ -289,6 +295,37 @@ def _linearize(self, jac=None, sub_do_ln=True):
         if (jac is None or jac is self._assembled_jac) and self._assembled_jac is not None:
             self._assembled_jac._update(self)
 
+    def add_output(self, name, val=1.0, **kwargs):
+        """
+        Add an output variable to the component.
+
+        Parameters
+        ----------
+        name : str
+            Name of the variable in this component's namespace.
+        val : float or list or tuple or ndarray
+            The initial value of the variable being added in user-defined units. Default is 1.0.
+        **kwargs : dict
+            Keyword args to store.  The value corresponding to each key is a dict containing the
+            metadata for the input name that matches that key.
+
+        Returns
+        -------
+        dict
+            Metadata for added variable.
+        """
+        metadata = super().add_output(name, val, **kwargs)
+
+        if "tags" in metadata:
```

Comment:
You don't need to do any of this checking here.  The base class calls `make_set` on tags, so you're guaranteed that tags exists and is a set by the time it gets here.  You can just do `metadata['tags'].add('openmdao:allow_desvar')` and that's it.

Reply:
Fixed